xdsl/utils/deprecation.py

```diff
@@ -1,30 +0,0 @@
-import warnings
```

Comment:
Wait, why removing this?
We should keep it for the next thing we deprecate, rather than having to find it again in the commits and adding it back.

Reply:
We've already migrated to using `deprecated` from `typing_extension`

Reply:
In fact it's already unused, all the ones I removed were not calling this function, and there are some remaining

Reply:
`deprecated` has been deprecated